projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -32,179 +39,243 @@
   }
 
-  private void createPdf() {
-    // TODO: Success reading license file only from "res/raw". Perhaps should be read from different
-    // dir
-    // Activate iText license
-    LicenseKey.loadLicenseFile(getResources().openRawResource(R.raw.itextkey));
+  // Save PDF to device. Return absFilename of the file
+  private String createPdf() {
+
+    // Read and fill template
+    String template = null;
     try {
-      HtmlConverter.convertToPdf(
-          fillTemplate(),
-          new FileOutputStream(Paths.get(lawsuitOutputPath, getPdfFilename()).toString()));
-      Toast.makeText(LawsuitPdfActivity.this, "Lawsuit created!", Toast.LENGTH_LONG).show();
+      template = fillTemplate();
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
 
-      // Read template / write output
+    PdfDocument document = new PdfDocument();
+
+    // Create 2 pages for the PDF (Size A4)
+    PdfDocument.PageInfo firstPageInfo = new PdfDocument.PageInfo.Builder(595, 842, 1).create();
+    PdfDocument.PageInfo secondPageInfo = new PdfDocument.PageInfo.Builder(595, 842, 2).create();
+    PdfDocument.Page firstPage = document.startPage(firstPageInfo);
+    PdfDocument.Page secondPage = null;
+    Canvas canvas = firstPage.getCanvas();
+    Paint paint = new Paint();
+    int rowCounter = 0;
+
+    int xPainter = 0;
+    int yPainter = 0;
+
+    // Draw lines to PDF
+    for (String line : template.split("\n")) {
+      // Switch to 2nd page on row 56
+      if (rowCounter == 56) {
+        document.finishPage(firstPage);
+        secondPage = document.startPage(secondPageInfo);
+        canvas = secondPage.getCanvas();
+        yPainter = 0;
+      }
+      // Draw text as RTL, 20 is for right padding
+      xPainter = (firstPageInfo.getPageWidth() - (int) paint.measureText(line)) - 20;
+      canvas.drawText(line, xPainter, yPainter, paint);
+      yPainter += paint.descent() - paint.ascent();
+      rowCounter++;
+    }
+    document.finishPage(secondPage);
+
+    // Save file to external storage
+    String absFilename =
+        Paths.get(
+                Environment.getExternalStorageDirectory().getPath(),
+                getString(R.string.app_name),
+                getString(R.string.output_folder_name),
+                (DATE_TIME_FORMATTER.format(new Date()) + ".pdf"))
+            .toString();
+    try {
+      document.writeTo(new FileOutputStream(new File(absFilename)));
+      Toast.makeText(this, "Done", Toast.LENGTH_LONG).show();
     } catch (IOException e) {
-      Log.w(TAG, "Read template / Write file: " + e.getMessage());
       e.printStackTrace();
-      Toast.makeText(LawsuitPdfActivity.this, "Failed to create lawsuit.", Toast.LENGTH_LONG)
-          .show();
+      Toast.makeText(this, "Something wrong: " + e.toString(), Toast.LENGTH_LONG).show();
     }
+
+    document.close();
+    return absFilename;
   }
 
+  // Return template filled with user and spam company's details
```

Comment:
Again, don't use code language. I would change it to something like "Fill the template with the user's and the company's details".

Reply:
Done.

Though I actually added this comment mainly for the purpose of explaining what exactly the function returns, not for what it does.

This I assumed is understood already from the function's name.